src/backend/dx12/src/lib.rs

```diff
@@ -1047,6 +1047,23 @@ impl hal::Instance<Backend> for Instance {
             //TODO: find a way to get a tighter bound?
             let sample_count_mask = 0x3F;
 
+            // Theoretically vram limited, but in practice 2^20 is the limit
+            let tier3_practical_descriptor_limit = 1 << 20;
+
+            let full_heap_count = match features.ResourceBindingTier {
+                d3d12::D3D12_RESOURCE_BINDING_TIER_1 => d3d12::D3D12_MAX_SHADER_VISIBLE_DESCRIPTOR_HEAP_SIZE_TIER_1,
+                d3d12::D3D12_RESOURCE_BINDING_TIER_2 => d3d12::D3D12_MAX_SHADER_VISIBLE_DESCRIPTOR_HEAP_SIZE_TIER_2,
+                d3d12::D3D12_RESOURCE_BINDING_TIER_3 => tier3_practical_descriptor_limit,
+                _ => unreachable!(),
```

Comment:
this isn't really unreachable, i.e. a later DX version may report TIER_4

Reply:
Alright, I can default it to be the same as 3 for now.